phpBB/phpbb/avatar/driver/upload.php

```diff
@@ -167,17 +167,27 @@ public function process_form($request, $template, $user, $row, &$error)
 			$destination = '';
 		}
 
+		$filedata['filename'] = $file->get('filename');
```

Comment:
Please change this to array notation, i.e.:
```
$filedata = array(
	'filename'		=> $file->get('filename'),
	'filesize'		=> $file->get('filesize'),
...
);
```

Also, since filedata won't be used after the event, please unset() the filedata array afterwards.